libretroshare/src/pqi/pqistreamer.cc

```diff
@@ -263,10 +279,11 @@ void pqistreamer::updateRates()
 			setRate(false, 0);
 		}
 
+		mAvgLastUpdate = t;
+		mAvgReadCount = 0;
+
 		{
```

Comment:
I really dont get the point in moving the lock here.

Reply:
mAvgLastUpdate and mAvgReadCount don't need to be mutex protected: they are both read and write accessed only by our thread (pqi).

The case of mAvgSentCount is a little bit particular. It is also read and write accessed only by our thread (pqi) and therefore does not need mutex protection. But in pqistreamer::tick_send you will see that the whole handleoutgoing_locked() process (the part that pops data from the mOutPkts outqueue and pass it to SSL) is unchanged: I left it under global mutex protection. It would have been possible and more effective (but also more complicated) to lock the mutex only when actually poping data from the mOutPkts outqueue, but I decided to leave for later. Thats means that the write access to mAvgSentCount when sending data is done by a "locked" method. For consistency I decided to protect also the reset of mAvgSentCount in updateRates even if it is actually not needed. I should probably have asked you what the good practice would have been here. We can change that of course and remove the lock.